common/include/pcl/common/impl/centroid.hpp

```diff
@@ -56,27 +56,30 @@ template <typename PointT, typename Scalar> inline unsigned int
 compute3DCentroid (ConstCloudIterator<PointT> &cloud_iterator,
                    Eigen::Matrix<Scalar, 4, 1> &centroid)
 {
-  // Initialize to 0
-  centroid.setZero ();
-
   unsigned cp = 0;
 
+  // Initialize to 0
+  Eigen::Matrix<Scalar, 4, 1> accu = Eigen::Matrix<Scalar, 4, 1>::Zero ();
+
   // For each point in the cloud
   // If the data is dense, we don't need to check for NaN
   while (cloud_iterator.isValid ())
   {
     // Check if the point is invalid
     if (pcl::isFinite (*cloud_iterator))
     {
-      centroid[0] += cloud_iterator->x;
-      centroid[1] += cloud_iterator->y;
-      centroid[2] += cloud_iterator->z;
+      accu += cloud_iterator->getVector4fMap ().template cast<Scalar> ();
       ++cp;
     }
     ++cloud_iterator;
   }
-  centroid /= static_cast<Scalar> (cp);
-  centroid[3] = 1;
+  if (cp)
```

Comment:
TBH I would prefer `cp != 0` here: cp is a number, not a boolean, and IMO it is better and clearer to not rely on 0 ~ false. Same in other places.

Reply:
I would usually use `cp != 0` as well. I think there was an earlier review that suggested this. I think as long as it's consistent, both are fine though. What is the preferred way for the library?

Reply:
I would go for `cp > 0` as it its initialized as 0 and eventually incremented.

Reply:
done